### ISSUE TITLE
Almost works but: Failed to register a ServiceWorker

### ISSUE THREAD

**Issue by kristianmandrup** [tony19/generator-polymer-init-2-x-app#2]
Hi Tony, tried your new version and much improved recipe.

```
$ polymer build
info:    Clearing build/ directory...
info:    (default) Building...
info:    (default) Build complete!
✔ ~/repos/polymer-projs/psk 
00:07 $ polymer serve -o build/default
info:    Files in this directory are available under the following URLs
      applications: http://127.0.0.1:8081
      reusable components: http://127.0.0.1:8081/components/polymer-starter-kit/
```

Now it build successfully :) However... when launched in the browser at http://127.0.0.1:8081/components/polymer-starter-kit/ I get the following browser console errors and blank screen

```
Uncaught (in promise) TypeError: Failed to register a ServiceWorker: A bad HTTP response code (404) was received when fetching the script.
webcomponents-lite.js:76 Uncaught TypeError: Cannot read property 'localName' of undefined
```

If I go to http://127.0.0.1:8081/

```
A bad HTTP response code (404) was received when fetching the script.
(index):1 Uncaught (in promise) TypeError: Failed to register a ServiceWorker: A bad HTTP response code (404) was received when fetching the script.
```

Could be related to this [issue](https://github.com/w3c/ServiceWorker/issues/915) ??

**Comment by kristianmandrup** [comment]
When I comment out the Service Worker registration: `// navigator.serviceWorker.register('service-worker.js');`

I still get this error: 

```
TypeError: Cannot read property 'localName' of undefined
    at S (webcomponents-lite.js:76)
```

How do you make it work?

**Comment by tony19** [comment]
> However... when launched in the browser at http://127.0.0.1:8081/components/polymer-starter-kit/ I get the following browser console errors and blank screen
>
> ```
Uncaught (in promise) TypeError: Failed to register a ServiceWorker: A bad HTTP response code (404) was received when fetching the script.
webcomponents-lite.js:76 Uncaught TypeError: Cannot read property 'localName' of undefined
```

You shouldn't get a blank screen for long. I noticed it takes a couple seconds before initially loading for some reason, but you should hit the Polymer 404 page:
![screen shot 2017-03-04 at 7 30 53 pm](https://cloud.githubusercontent.com/assets/26580/23583673/1ff607ce-0111-11e7-9d49-fc9753f380f4.png)

That's because Polymer CLI serves up a default component path, which doesn't apply to the starter-kit. The starter-kit has client-side routing (via `<app-route>`) that attempts to switch to a page, identified by the first path component in the URL (i.e., "components"). Since there is no "components" page, the app falls back to its custom 404 page.

Regarding the service-worker error message, I'm not able to reproduce that error. You might be running off a previously installed service worker. I recommend clearing cache or using an incognito window.

**Comment by tony19** [comment]
Oh, but I noticed some missing project files in the generated output. Will fix.

**Comment by tony19** [comment]
Fixed in [0.0.8](https://github.com/tony19/generator-polymer-init-2-x-app/releases/tag/v0.0.8)

**Comment by kristianmandrup** [comment]
Sweet :)
